src/biokbase/narrative/jobs/jobmanager.py

```diff
@@ -630,6 +630,20 @@ def cancel_job(self, job_id: str, parent_job_id: str = None) -> None:
                 self._running_jobs[job_id]["refresh"] = is_refreshing
                 del self._running_jobs[job_id]["canceling"]
 
+    def retry_job(self, job_id: str):
+        if job_id is None:
+            raise ValueError("Job id required for retrying")
+        if job_id not in self._running_jobs:
+            raise ValueError(f"No job present with id {job_id}")
```

Comment:
Would the error be "Cannot retry a running job?"

Reply:
Actually, the job must NOT be running in order to retry it.

Reply:
I think we were going to just check that the job didn't have state 'completed' -- anything else would be fair game as it's possible that the job could have changed status since the last time the backend polled ee2 for the job status.

Reply:
@bio-boris  is there an ee2 endpoint where one can submit a job ID and see if it has a parent ID?

Reply:
@ialarmedalien you can use check jobs